spotify.go

```diff
@@ -124,6 +171,78 @@ func decodeError(r io.Reader) error {
 // authenticate, you can use `DefaultClient`.
 type Client struct {
 	http *http.Client
+
+	autoRetry     bool
+	retryDuration time.Duration
+}
+
+func (c *Client) SetAutoRetry(flag bool) {
+	c.autoRetry = flag
+}
+
+// executeOpt executes a non-GET request. `needsStatus` describes another code
+// that can represent success. Note that in all current usages of this function,
+// we need to still allow a 200 even if we'd also like to check for a second
+// success code.
+func (c *Client) executeOpt(req *http.Request, needsStatus int, result interface{}) error {
```

Comment:
I think I would like to try a new pattern here.  Let's remove `executeOpt` and have a single `execute` method that looks like this:

```
func (c *Client) execute(req *http.Request, result interface{}, needsStatus ...int) error
```

Then inside execute, if `len(needsStatus) > 0` and the status code was not a 200 OK, you can check if the status code is found in the needsStatus slice.

The advantages to this are:
- only a single function that works with and without the optional status parameter(s)
- support for multiple valid status codes